### PR TITLE
PR for Feature/friendship v1

### PR DESCRIPTION
**Pull request by Lameck1**
This PR submits feature for friendship-v1

### Task completed
As a guest user:

I want to be able to send a friendship invitation.
I want to see a button “Add Friend” next to the name of user who is not my friend yet - on both users’ list and single-user page.
I want to be able to see pending friendship invitations sent to me from other users.
I want to be able to accept or reject friendship's invitation.

### CODE REVIEW

**Review by rindrajosia — CHANGES_REQUESTED** [review]
Invalid :red_circle: 

Hey team, please activate Stickler in your pull request. This code review will be mark as invalid in your dashboard :cry: 

That's it, you can submit another code review request after making these changes
With all my respect
Rindra Josia

**Comment by Lameck1** [comment]
Thanks, @rindrajosia, that passed my thought.

**Review by mke2111 — CHANGES_REQUESTED** [review]
# **STATUS: CHANGES REQUIRED 🔴 ⚠️** 

## Hey @Lameck1, 👋🏽 
 **Thank you for the good job you got going on. The app works just fine.** 👌🏽 
However, 🛑  there are a few changes to make to meet the minimum requirements for approval.

## REQUIRED CHANGES

- [x] Posts timeline should include posts written by logged-in user and their friends ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/)). Right now, the timeline includes posts from all users, even when there is no friendship.
- [x] Well done with the association tests added. Now, you need to add more tests ( feature integration tests & request integration tests). You may choose to use Capybara or Factory gem.
- [x] Remove all logic from the views files. These files are only supposed to render the pages linked to the respective controllers and as such, all logic should be in from the controllers or helpers. You may now safely transfer all logic (conditional statements and loops) to the helpers.

## OPTIONAL CHANGES ⚪ 

- [x] You may now add the Authors to the readme file.


**Kindly make all the requested changes and submit another CRR. 👍🏽** 

## Cheers!! 🍹 & Happy Coding.


### Ping me on slack for a  🆘  **Roy Mukuye**
### Mention me [@mke2111](https://github.com/mke2111) in this PR and I'll come through for your inquiries.

File: app/views/users/_user.html.erb [hunk 1]
```diff
@@ -1,6 +1,9 @@
 <li>
-  Name: <%= user.name %>
-  <span class="profile-link">
-    <%= link_to 'See Profile',  user_path(user), class: 'profile-link' %>
-  </span>
+  <% if user != current_user %>
```
Comment (mke2111) [inline app/views/users/_user.html.erb]: There should be NO logic in the views folder

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -1,5 +1,18 @@
 <section class="user-section">
   <h2><%= "Name: #{@user.name}" %></h2>
+  <% if current_user != @user %>
```
Comment (mke2111) [inline app/views/users/show.html.erb]: There should be NO logic in the views folder

File: app/views/users/show.html.erb [hunk 3]
```diff
@@ -1,5 +1,18 @@
...
+    <%= friendship_button(@user) %>
+  <% else %>
+    <div>
+      <h4>Friendships Requests:</h4>
+      <ul>
+       <% @user.friend_requests.each do |u| %>
```
Comment (mke2111) [inline app/views/users/show.html.erb]: There should be NO logic in the views folder

**Review by GzimAsani — CHANGES_REQUESTED** [review]
# Status: Changes Required

Hello team, good job implementing the changes asked from the previous reviewer, but you still have to implement the following request asked previously:

- [x] _Posts timeline should include posts written by logged-in user and their friends ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/)). Right now, the timeline includes posts from all users, even when there is no friendship._

### Please implement these changes and ask for another code review whenever you are ready

_**If you have any question related to the required changes feel free to text me on slack @GzimAsani** or just add a comment here_

HAPPY CODING :computer: :desktop_computer:

**Review by umairarshadbutt — APPROVED** [review]
# **Status: PROJECT APPROVED** :champagne:
Excellent job on this project guys. :fireworks: Keep it up.
##
**All the required tasks have been fulfilled in this project. I will go ahead and mark this as complete.**
- Congratulations on the splendid job @Lameck1 ! :+1: 
- You can now merge to master and submit the project completion form :rocket: . I wish you all the very best in your next tasks :1st_place_medal:
##
**If you have any questions, let me know by leaving a comment here and mentioning me. I will be happy to help**
HAPPY CODING :computer:
**[Umair Arshad](https://umairarshad.me/)**
[Twitter](https://twitter.com/its_UmairArshad) | [LinkedIn](https://www.linkedin.com/in/umair-arshad-butt/) | [GitHub](https://github.com/umairarshadbutt)

**Pull request merged**